IntegrationTest Failures for Vault 1.9.0, 1.10, and 1.11

New IntegrationTest failures were introduced in Vault versions 1.9.0, 1.10, and 1.11. Creating an issue to track all of the issues in test_integration.py.

Failures introduced in 1.9.0
```
FAILED tests/integration_tests/v1/test_integration.py::IntegrationTest::test_auth_kubernetes - AssertionError: 'claim "iss" is invalid' not found in 'open /var/run/secrets/kubernetes.io/serviceaccount/ca.crt: no such file or directory, on post https://localhost:8200/v1/auth/k8s/login'
```

Failures introduced in 1.10
```
FAILED tests/integration_tests/v1/test_integration.py::IntegrationTest::test_auth_gcp_alternate_mount_point_with_no_client_token_exception - hvac.exceptions.Forbidden: permission denied, on post https://localhost:8200/v1/auth/gcp/login
```

Failures introduced in 1.11
```
FAILED tests/integration_tests/v1/test_integration.py::IntegrationTest::test_create_kubernetes_configuration - hvac.exceptions.InternalServerError: 1 error occurred:
FAILED tests/integration_tests/v1/test_integration.py::IntegrationTest::test_create_kubernetes_role - hvac.exceptions.InternalServerError: 1 error occurred:
FAILED tests/integration_tests/v1/test_integration.py::IntegrationTest::test_delete_kubernetes_role - hvac.exceptions.InternalServerError: 1 error occurred:
FAILED tests/integration_tests/v1/test_integration.py::IntegrationTest::test_get_kubernetes_configuration - hvac.exceptions.InternalServerError: 1 error occurred:
FAILED tests/integration_tests/v1/test_integration.py::IntegrationTest::test_get_kubernetes_role - hvac.exceptions.InternalServerError: 1 error occurred:
FAILED tests/integration_tests/v1/test_integration.py::IntegrationTest::test_list_kubernetes_roles - hvac.exceptions.InternalServerError: 1 error occurred:
```

#868 Deals with most of these tests.

The Kubernetes tests have been updated to use the auth.kubernetes module and moved out of the integration tests.

The only test that's left to deal with is 

```
FAILED tests/integration_tests/v1/test_integration.py::IntegrationTest::test_auth_gcp_alternate_mount_point_with_no_client_token_exception - hvac.exceptions.Forbidden: permission denied, on post https://localhost:8200/v1/auth/gcp/login
```

I'll try to look into this today and see if I can figure out what's wrong.

--- a/tests/integration_tests/v1/test_integration.py
+++ b/tests/integration_tests/v1/test_integration.py
@@ -489,10 +489,19 @@ class IntegrationTest(HvacIntegrationTestCase, TestCase):
 
         # When attempting to auth (POST) to an auth backend mounted at a different path than the default, we expect a
         # generic 'missing client token' response from Vault.
-        with self.assertRaises(exceptions.InvalidRequest) as assertRaisesContext:
+        expected_exception = (
+            exceptions.InvalidRequest
+            if utils.vault_version_lt("1.10.0")
+            else exceptions.Forbidden
+        )
+        with self.assertRaises(expected_exception) as assertRaisesContext:
             self.client.auth.gcp.login("example-role", jwt)
 
-        expected_exception_message = "missing client token"
+        expected_exception_message = (
+            "missing client token"
+            if utils.vault_version_lt("1.10.0")
+            else "permission denied"
+        )
         actual_exception_message = str(assertRaisesContext.exception)
         self.assertIn(expected_exception_message, actual_exception_message)
 

